### DISCUSSION TITLE
Renovate does not detect all updates in a Drupal project using composer

### DISCUSSION THREAD

**Discussion by stmh** [renovatebot/renovate#15203] [Request Help] [answered]
Before creating an issue I wanted to get first some pointers, how I can debug the following issue:

I am using a self hosted renovate installation (32.x) interfacing with a self-hosted gitlab installation. renovates checks several Drupal projects and I noticed that one Drupal project detects new Drupal core versions, the other not. Both projects share the same renovate config.

```
 {
                 "depType": "require",
                 "depName": "drupal/core-recommended",
                 "currentValue": "^9",
                 "datasource": "packagist",
                 "lockedVersion": "9.3.9",
                 "depIndex": 6,
                 "updates": [],
                 "warnings": [],
                 "versioning": "composer",
                 "sourceUrl": "https://github.com/drupal/core-recommended",
                 "currentVersion": "9.3.9",
                 "fixedVersion": "9.3.9"
               },
```

(notice the missing `updates`-section here, In the other project I can see the actual updates.

Running `composer outdated | grep drupal` shows the new versions.

What's the best way to debug this issue? Increasing the log level to debug does not give me any meaningful pointers.

thanks in advance,

Stephan

**Comment by rarkins** [comment, marked as answer]
Read the documentation for `rangeStrategy`, in particular the setting `update-lockfile`

**Reply by stmh** [reply]
thanks, using `update-lockfile` seems to fix the problems. This side effect was not clear for me, when I was reading the docs initially.
